test/test_dpres_plevel.py

```diff
@@ -0,0 +1,140 @@
+                            40., 30., 25., 20., 10.]) # units hPa
+# convert hPa to Pa
+pressure_levels = pressure_levels * 100.0
+
+pressure_levels_asfloat32 = pressure_levels.astype(np.float32)
+
+# Surface pressure (scalar and 2D, as well as 2D with np.nan and -99 missing values)
+pressure_surface_scalar = 101800.0   # Units of Pa
+pressure_surface_2d = np.array([1018.0, 1016.0, 1014.0, 1012.0]).reshape(2, 2)   # Units of Pa
+pressure_surface_2d = pressure_surface_2d * 100
+
+pressure_surface_2d_asfloat32 = pressure_surface_2d.astype(np.float32)
+
+# missing value of np.nan
+pressure_surface_2d_nan = pressure_surface_2d.copy()
+pressure_surface_2d_nan[0, 1] = np.nan
+
+pressure_surface_2d_nan_asfloat32 = pressure_surface_2d_nan.astype(np.float32)
+
+# missing value of -99
+pressure_surface_2d_msg = pressure_surface_2d_nan.copy()
+pressure_surface_2d_msg[np.isnan(pressure_surface_2d_msg)] = -99.0
+
+pressure_surface_2d_msg_asfloat32 = pressure_surface_2d_msg.astype(np.float32)
+
+# Expected Output
+# for scalar pressure_surface
+expected_dp_psfc_scalar = np.array([4300., 5000., 5000., 5000., 5000., 5000., 5000., 5000.,
+                                    5000., 5000., 5000., 5000., 5000., 5000., 5000., 5000.,
+                                    3750., 2500., 2500., 2500., 2250., 1500., 1000., 1000.,
+                                    1000., 1000., 750., 500., 750., 500.])
+
+# for 2D pressure_surface
+expected_dp_psfc_2d     = np.array([4300., 4100., 3900., 3700., 5000., 5000., 5000., 5000.,
+                                    5000., 5000., 5000., 5000., 5000., 5000., 5000., 5000.,
+                                    5000., 5000., 5000., 5000., 5000., 5000., 5000., 5000.,
+                                    5000., 5000., 5000., 5000., 5000., 5000., 5000., 5000.,
+                                    5000., 5000., 5000., 5000., 5000., 5000., 5000., 5000.,
+                                    5000., 5000., 5000., 5000., 5000., 5000., 5000., 5000.,
+                                    5000., 5000., 5000., 5000., 5000., 5000., 5000., 5000.,
+                                    5000., 5000., 5000., 5000., 5000., 5000., 5000., 5000.,
+                                    3750., 3750., 3750., 3750., 2500., 2500., 2500., 2500.,
+                                    2500., 2500., 2500., 2500., 2500., 2500., 2500., 2500.,
+                                    2250., 2250., 2250., 2250., 1500., 1500., 1500., 1500.,
+                                    1000., 1000., 1000., 1000., 1000., 1000., 1000., 1000.,
+                                    1000., 1000., 1000., 1000., 1000., 1000., 1000., 1000.,
+                                    750.,  750.,  750.,  750.,  500.,  500.,  500.,  500.,
+                                    750.,  750.,  750.,  750.,  500.,  500.,  500.,  500.]).reshape(30,2,2)
+
+# for msg_py = -np.nan input
+expected_dp_psfc_2d_msg_nan = expected_dp_psfc_2d.copy()
```

Comment:
to confirm, we are setting all the pressures to np.nan at just this one point on the grid?

Reply:
More precisely: The `expected_dp_psfc_2d_msg_nan` is the expected output when we give a 2D surface pressure input (2x2) with only one element is `np.nan`, i.e. `pressure_surface_2d_nan` . In such a case, all the pressures at just one point on the grid are expected to be `np.nan` as you pointed out.

Reply:
Thanks! Marking as resolved.